src/metabase/driver.clj

```diff
@@ -446,36 +446,41 @@
-  `:named` clause:
-
-    [:named [:count] \"My Count\"]
-
-  This name was used for both the `:display_name` in the query results, and for the `:name` used as an alias in the
-  query (e.g. the right-hand side of a SQL `AS` expression). Because some custom display names weren't allowed by some
-  drivers, or would be transformed in some way (for example, Redshift always lowercases custom aliases), this method
-  was needed so we could match the name we had given the column with the one in the query results.
-
-  In 0.33.0, we started using `:named` internally to alias aggregations in middleware in *all* queries to prevent
-  issues with referring to multiple aggregations of the same type when that query was used as a source query.
-  See [#9767](https://github.com/metabase/metabase/issues/9767) for more details. After this change, it became
-  desirable to differentiate between such internally-generated aliases and display names, which need not be used in
-  the query at all; thus in MBQL 1.3.0 [`:named` was replaced by the more general
-  `:aggregation-options`](https://github.com/metabase/mbql/pull/7). Because user-generated names are no longer used as
-  aliases in native queries themselves, this method is no longer needed and will be removed in a future release."
+(defmulti ^{:deprecated "0.42.0"} format-custom-field-name
+  "Unused in Metabase 0.42.0+. Implement [[escape-alias]] instead. This method will be removed in a future release."
   {:arglists '([driver custom-field-name])}
   dispatch-on-initialized-driver
   :hierarchy #'hierarchy)
 
-(defmethod format-custom-field-name ::driver [_ custom-field-name]
-  custom-field-name)
+(defmulti ^String escape-alias
+  "Escape a dynamically generated `column-alias-name` string in a way that makes it valid for your database. These
+  column aliases are generated in [[metabase.query-processor.util.add-alias-info]] and elsewhere for existing columns,
+  aggregate functions and other expressions, and for columns from source queries or joins. For `:sql` drivers, the
+  aliases generated here will be quoted in the resulting SQL.
+
+  The default impl of [[escape-alias]] returns the column alias as-is. You may override this to escape names that
+  your particular database may not allow (even when quoted).
+
+  Drivers that do not use the info generated by [[metabase.query-processor.util.add-alias-info]] do not need to
+  implement this method. The `:sql` driver *DOES* use this information, so `:sql` drivers should implement this
+  method if they need to do something specific."
+  {:added "0.42.0", :arglists '([driver column-alias-name])}
+  dispatch-on-initialized-driver
+  :hierarchy #'hierarchy)
+
+;; TODO -- I have a PR to add some truncation logic for this so we don't end up with crazy-long identifiers -- see
+;; #19659
+;;
+;; TODO -- we should probably also remove diacritical marks like we do for BigQuery
+(defmethod escape-alias ::driver
+  [_driver alias-name]
```

Comment:
Is the first param supposed to be either `_` or `driver` ?

Reply:
This is intentional. See https://guide.clojure.style/#naming-unused-bindings

> However, when it can help the understanding of your code, it can be useful to explicitly name unused arguments or maps you’re destructuring from. In this case, prepend the name with an underscore to explicitly signal that the variable is supposed to be unused.